### PR TITLE
remove string_view quote

### PR DESCRIPTION
**Pull request by neheb**
It's not clear that it's a good idea.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by whyman** [comment]
Why is it not a good idea?

**Comment by KarlStraussberger** [comment]
It used to be const char*. Actually, I prefer the `string_view` because of the comparison operators.

**Comment by neheb** [comment]
sure. but there's no point here. All usages use the data() method.

File: src/database/sql_database.h [hunk 1]
```diff
@@ -96,10 +96,9 @@ class SQLDatabase : public Database {
     /* methods to override in subclasses */
     virtual std::string quote(const std::string& str) const = 0;
     /* wrapper functions for different types */
-    std::string quote(std::string_view str) const { return quote(std::string(str)); }
-    // required to handle #defines
     std::string quote(const char* str) const { return quote(std::string(str)); }
     std::string quote(char val) const { return quote(fmt::to_string(val)); }
+    // required to handle #defines
```
Comment (KarlStraussberger) [inline src/database/sql_database.h:101]: The comment belongs to the `const char*` version

Comment (neheb) [reply src/database/sql_database.h:101]: Interesting... Wonder if it makes sense to replace those defines with string_view instead of this PR.

Comment (neheb) [reply src/database/sql_database.h]: hmm if I remove that const char version, it still compiles.

**Pull request merged**